priiloader/source/main.cpp

```diff
@@ -1592,6 +1683,73 @@ void BootMainSysMenu( void )
 			} //end for loop of all patches of hack[i]
 		} // end general hacks loop*/
 
+		// Apply vWii specific patches which are always required for the menu to boot from priiloader
+		if (smTmd->vwii_title)
+		{
+			gprintf("applying vWii patches");
+
+			// Patch out IPC wait
+			static u32 ipc_wait_hash[] = { 0x546007bd, 0x4182fff4 };

```

Comment:
isn't it possible to add it to the list of hacks it needs to apply and let the rest of the code apply it? that way youre not looping over the memory 3 or 4 times per boot :)

Reply:
Haven't looked too much into how hacks are implemented, but it seemed like the `system_hacks` were directly tied to the hashes of which hacks are enabled, so I tried to avoid touching that.
It seems like each hack iterates over the entire system memory binary anyways?

Reply:
checked, i thought i designed it to loop memory and check every hack on every memory address. whoops.
would it be more efficient/cleaner if there was a new vector with the enabled hacks + the added ones and then have the one loop dealing with all of them?

Reply:
Yeah, that would probably make it easier to add built-in hacks for specific versions of the system menu.

Reply:
Going to mark this as resolved for now. Should be fine for now until this part is changed.